### PR TITLE
Add helper methods to clear caches cleanly

### PR DESCRIPTION
**Pull request by samuelhwilliams**
## 🎫 Ticket
BAU

## 📝 Description
While working on [the persisted submission status](https://github.com/communitiesuk/funding-service/pull/1681) PR, we noticed that cache clearing was quite manual and fragmented.

This adds helpers to our classes that cache some properties to reliably clear all cached properties and updates the tests to stop just poking specific cached properties.

There were also a lot of tests that were just unnecessarily clearing caches.

### Review Readiness
- [x] PR title and description provide sufficient context for reviewers
- [x] Commits are logical, self-contained, and have clear descriptions

### Testing
- [x] I have tested this change and it meets the acceptance criteria for the ticket
- [ ] I need the reviewer(s) to pull and run this change locally
- [x] New (non-developer) functionality has appropriate unit and integration tests
- [ ] End-to-end tests have been updated (if applicable)
- [ ] Edge cases and error conditions are tested

### CODE REVIEW

File: app/common/data/models.py [hunk 1]
```diff
@@ -454,6 +454,12 @@ class Form(BaseModel):
         cascade="all, save-update, merge",
     )
 
+    def clear_caches(self):
+        for attr, value in self.__class__.__dict__.items():
+            if isinstance(value, cached_property):
+                if attr in self.__dict__:
+                    del self.__dict__[attr]
+
```
Comment (srh-sloan) [inline app/common/data/models.py:539]: Could we add this to the BaseModel and then it's there for any models that need it in future?

**Review by srh-sloan — COMMENTED** [review]
One comment, fine to ignore if we've already thought about it and decided not to 👍

File: app/common/data/models.py (hunk 1 above)
Comment (samuelhwilliams) [reply app/common/data/models.py:539]: I think I'd vaguely decided that having a `clear_cache` thing on things with no cache was maybe confusing/misleading. We could potentially add a mixin but I think it's fine to leave for a future need

**Review by srh-sloan — APPROVED** [review]
Much nicer to have this all tidied up and in one place 🎉

File: app/deliver_grant_funding/templates/deliver_grant_funding/reports/list_submissions.html [hunk 2]
```diff
@@ -132,9 +132,9 @@ <h1 class="govuk-heading-l">
         {% endif %}
 
         {% if submission.submission_id %}
-          {% do cells.append({"html": submissionStatusTag(submission.status, report.is_closed and not submission.status == enum.submission_status.SUBMITTED)}) %}
+          {% do cells.append({"html": submissionStatusTag(submission.status, report.is_overdue and not submission.status == enum.submission_status.SUBMITTED)}) %}
         {% else %}
-          {% do cells.append({"html": submissionStatusTag(enum.submission_status.NOT_SUBMITTED if report.is_closed else enum.submission_status.NOT_STARTED, report.is_closed)}) %}
+          {% do cells.append({"html": submissionStatusTag(enum.submission_status.NOT_SUBMITTED if report.is_overdue else enum.submission_status.NOT_STARTED, report.is_closed)}) %}
         {% endif %}
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply app/deliver_grant_funding/templates/deliver_grant_funding/reports/list_submissions.html:138]: being handled by @albertkol
